rs/rosetta-api/icrc1/ledger/sm-tests/src/lib.rs

```diff
@@ -3980,6 +3999,27 @@ test_bytes";
         "Expected: {}, got: {}",
         expected_message, message
     );
+
+    args.arg = Encode!(&ApproveArgs {
+        from_subaccount: None,
+        ..approve_args.clone()
+    })
+    .unwrap();
+    args.user_preferences.metadata.utc_offset_minutes = None;
+
+    let message = extract_icrc21_message_string(
+        &icrc21_consent_message(env, canister_id, Principal::anonymous(), args.clone())
+            .unwrap()
+            .consent_message,
+    );
+
+    let expected_message = expected_approve_message.replace("\n\n**Your account:**\nd2zjj-uyaaa-aaaaa-aaaap-4ai-qmfzyha.101010101010101010101010101010101010101010101010101010101010101","\n\n**Your Subaccount:**\n0000000000000000000000000000000000000000000000000000000000000000" )
```

Comment:
This test is a bit cumbersome to read - perhaps it would help to split it into separate functions for the individual tests, and creating the `expected_message` from scratch (or using a builder?) instead of replacing strings in the `expected_approve_message` defined 100+ lines earlier?